CMSgov/mdct-mcr
# MDCT-2025: DynamicField Deletion Deletes Related Data

## Description

Closes [MDCT-2025](https://qmacbis.atlassian.net/browse/MDCT-2025).

`DynamicField` (i.e. plans or BSS entities) have related sanctions and quality measures responses. As it stands, these sanctions and quality measures responses aren't deleted when an entity is deleted, causing data mismatching in the DB.

With the changes in this PR, the associated sanctions and quality measures responses are now deleted at the same time the entity is deleted.

Changes:
- Modifies entity deletion method to include filtering sanction and quality measures data before setting new entity, sanction, and quality measure data all at once.
- Adds associated tests.
- Modifies backend validation schema (frontend validation schema change not necessary) to allow entities (plans, bss entities) to be set to `[]` so we can successfully delete all entities and return to initial state (fixes a previous bug).
- Minor tweaks related to ^^ to prevent console errors.


### How to test

1. Add some plans
    - Navigate to `/mcpar/program-information/add-plans`
    - Add two plans (mentally select one to delete and remember it for the next step)
2. Add some sanctions and quality measures
    - Navigate to `/mcpar/plan-level-indicators/sanctions`
    - Create a sanction (pick the plan you intend to delete) -- no need to fill out the data in the drawer
    - Navigate to `/mcpar/plan-level-indicators/quality-measures`
    - Create a quality measure and enter data in the drawer for each plan measure
3. Delete the plan and check
    - Navigate to `/mcpar/program-information/add-plans`
    - Delete the plan you selected earlier
    - Navigate to `/mcpar/plan-level-indicators/sanctions`. The sanction should be gone!
    - Navigate to `/mcpar/plan-level-indicators/quality-measures` and open the drawer. The response for the plan you delete should be gone!

### Changed Dependencies
N/A

## Code author checklist

## Reviewer checklist (two different people)
